.github/workflows/release.yaml

```diff
@@ -0,0 +1,191 @@
+        description: 'Publish as a draft'
+        required: true
+        default: true
+        type: boolean
+
+env:
+  DRAFT_RELEASE: ${{ github.event_name == 'push' && github.ref_name == 'ci-release-test' && 'true' || github.event_name == 'workflow_dispatch' && inputs.draft_release == true && 'true' || 'false' }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  run_if:
+    if: "${{ (github.event_name == 'push' && startsWith(github.event.head_commit.message, 'pre-release: Update version to')) || github.event_name == 'workflow_dispatch' }}"
+    runs-on: ubuntu-latest
+    outputs:
+      release_tag: ${{ steps.release_tag.outputs.release_tag }}
+    steps:
+      - run: echo "Triggered by ${{ github.event_name }}"
+      - run: 'echo "Draft release: ${{ env.DRAFT_RELEASE }}"'
+      - id: release_tag
+        run: |
+          if [[ ${{github.event_name}} == "push" ]]
+          then
+            [[ "${{github.event.head_commit.message}}" =~ ^pre-release:\ Update\ version\ to\ ([0-9]*\.[0-9]*\.[0-9]*(\-[0-9a-z]+)?)\ .*$ ]] && echo "release_tag=${BASH_REMATCH[1]}" >> $GITHUB_OUTPUT
+          else
+            echo "release_tag=${{inputs.release_tag}}" >> $GITHUB_OUTPUT
+          fi
+
+  create_tag:
+    needs: run_if
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    env:
+      RELEASE_TAG: ${{ needs.run_if.outputs.release_tag }}
+    outputs:
+      tag_url: ${{ steps.output_tag.outputs.tag_url }}
+    environment:
+      name: release
+      url: ${{ steps.output_tag.outputs.tag_url }}
+    steps:
+    - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+    - id: create_tag
+      run: |
+        git config --global user.name 'Kasten Production'
+        git config --global user.email 'infra@kasten.io'
+        git tag -a "${RELEASE_TAG}" -m "Release version"
+        git push origin "${RELEASE_TAG}"
+    - id: output_tag
+      run: echo "tag_url=https://github.com/kanisterio/kanister/releases/tag/${RELEASE_TAG}" >> "$GITHUB_OUTPUT"
+
+  release_packages:
+    runs-on: ubuntu-latest
+    needs: [run_if, create_tag]
+    permissions:
+      packages: write
+      contents: write
+    env:
+      RELEASE_TAG: ${{ needs.run_if.outputs.release_tag }}
+    outputs:
+      release: ${{ steps.output_release.outputs.release_url }}
+    environment:
+      name: release
+      url: ${{ steps.output_release.outputs.release_url }}
+    steps:
+    - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      with:
+        ref: ${{ env.RELEASE_TAG }}
+        fetch-depth: 0
+    - name: build helm charts
+      run: |
+        export PACKAGE_FOLDER=helm_package
+        export HELM_RELEASE_REPO_URL=https://github.com/kanisterio/kanister/releases/download/${RELEASE_TAG}
+        export HELM_RELEASE_REPO_INDEX=https://charts.kanister.io/
+        make package-helm VERSION=${RELEASE_TAG}
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@main
```

Comment:
Please tag this action's version. It should not use a "floating" tag or repo branch name.